project/GenerateAST.scala

```diff
@@ -0,0 +1,82 @@
+import java.io.IOException
+
+import sbt.Keys._
+import sbt._
+import sbt.internal.util.ManagedLogger
+
+import scala.sys.process._
+
+object GenerateAST {
+
+  val rustVersion = settingKey[String]("rustc version used in the project")
+  private val cargoCmd = "cargo"
+
+  lazy val task = Def.task {
+    val log    = state.value.log
+    val lib    = baseDirectory.value.getParentFile.getParentFile / "lib"
+    val source = lib / "rust/ast/src/ast.rs"
+    val output = lib / "scala/ast/src/main/scala/org/enso/ast/Ast.scala"
+    val cache  = streams.value.cacheStoreFactory.make("ast_source")
+
+    verifyRustVersion(rustVersion.value) match {
```

Comment:
This verification should take place as part of the global `EnvironmentCheck`, rather than only when the rust generation task is triggered.